File: Low Level Design (LLD)/DesignPatterns/3. Structural/Mediator/Java/ConcreteUser.java
public class ConcreteUser extends User {

    public ConcreteUser(ChatMediator mediator, String name) {
        super(mediator, name);
    }

    @Override
    public void send(String message) {
        System.out.println(this.name + " sends: " + message);
        mediator.showMessage(message, this);
    }

    @Override
    public void receive(String message, User sender) {
        System.out.println(this.name + " receives from " + sender.getName() + ": " + message);
    }
}

File: Low Level Design (LLD)/DesignPatterns/3. Structural/Mediator/Cpp/src/ConcreteUser.cpp
#include "../include/ConcreteUser.h"

ConcreteUser::ConcreteUser(ChatMediator* mediator, const std::string& name)
    : User(mediator, name) {}

void ConcreteUser::send(const std::string& message) {
    std::cout << name << " sends: " << message << std::endl;
    mediator->showMessage(message, this);
}

void ConcreteUser::receive(const std::string& message, User* sender) {
    std::cout << name << " receives from " << sender->getName() << ": " << message << std::endl;
}
